### ISSUE TITLE
Failed to inject frame into filter network: Invalid argument

### ISSUE THREAD

**Issue by littlewhitesea** [slhck/ffmpeg-quality-metrics#35]
**According to your detailed description, I have finised the installation of ffmpeg and ffmpeg_quality_metrics.

When I ran the following command, I can get the right output.
_ffmpeg_quality_metrics dis.mp4 ref.avi -m psnr ssim_

However, I met this an error when I performed this command.
_ffmpeg_quality_metrics dis.mp4 ref.avi -m vmaf_

The entire report of this error is shown in the following.**

Traceback (most recent call last):
  File "/media/data4/vqe/anaconda3/bin/ffmpeg_quality_metrics", line 8, in <module>
    sys.exit(main())
  File "/media/data4/vqe/anaconda3/lib/python3.8/site-packages/ffmpeg_quality_metrics/__main__.py", line 131, in main
    ffqm.calc(metrics, vmaf_options=vmaf_options)
  File "/media/data4/vqe/anaconda3/lib/python3.8/site-packages/ffmpeg_quality_metrics/ffmpeg_quality_metrics.py", line 290, in calc
    raise e
  File "/media/data4/vqe/anaconda3/lib/python3.8/site-packages/ffmpeg_quality_metrics/ffmpeg_quality_metrics.py", line 286, in calc
    output = self._run_ffmpeg_command(filter_chains, desc=", ".join(metrics))
  File "/media/data4/vqe/anaconda3/lib/python3.8/site-packages/ffmpeg_quality_metrics/ffmpeg_quality_metrics.py", line 507, in _run_ffmpeg_command
    _, stderr = run_command(cmd, self.dry_run, self.verbose)
  File "/media/data4/vqe/anaconda3/lib/python3.8/site-packages/ffmpeg_quality_metrics/utils.py", line 79, in run_command
    raise RuntimeError(
RuntimeError: error running command: ffmpeg -nostdin -nostats -y -threads 0 -r 25.0 -i ref.avi -r 25.0 -i dis.mp4 -filter_complex '[1][0]scale2ref=flags=bicubic[dist][ref];[dist]setpts=PTS-STARTPTS[distpts];[ref]setpts=PTS-STARTPTS[refpts];[distpts][refpts]libvmaf='"'"'model_path=/media/data4/vqe/anaconda3/lib/python3.8/site-packages/ffmpeg_quality_metrics/vmaf_models/vmaf_v0.6.1.json:phone_model=0:log_path=/tmp/5r8mcd96-vmaf.txt:log_fmt=json:psnr=1:ssim=1:ms_ssim=1:n_threads=32'"'"'' -an -f null /dev/null
ffmpeg version N-57736-ga501d55905-static https://johnvansickle.com/ffmpeg/  Copyright (c) 2000-2021 the FFmpeg developers
  built with gcc 8 (Debian 8.3.0-6)
  configuration: --enable-gpl --enable-version3 --enable-static --disable-debug --disable-ffplay --disable-indev=sndio --disable-outdev=sndio --cc=gcc --enable-fontconfig --enable-frei0r --enable-gnutls --enable-gmp --enable-libgme --enable-gray --enable-libaom --enable-libfribidi --enable-libass --enable-libvmaf --enable-libfreetype --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-librubberband --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libvorbis --enable-libopus --enable-libtheora --enable-libvidstab --enable-libvo-amrwbenc --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxml2 --enable-libdav1d --enable-libxvid --enable-libzvbi --enable-libzimg
  libavutil      57.  0.100 / 57.  0.100
  libavcodec     59.  1.101 / 59.  1.101
  libavformat    59.  3.100 / 59.  3.100
  libavdevice    59.  0.100 / 59.  0.100
  libavfilter     8.  0.102 /  8.  0.102
  libswscale      6.  0.100 /  6.  0.100
  libswresample   4.  0.100 /  4.  0.100
  libpostproc    56.  0.100 / 56.  0.100
Input #0, avi, from 'ref.avi':
  Metadata:
    software        : Lavf59.3.100
  Duration: 00:00:01.64, start: 0.000000, bitrate: 3920 kb/s
  Stream #0:0: Video: mpeg4 (Simple Profile) (FMP4 / 0x34504D46), yuv420p, 720x576 [SAR 1:1 DAR 5:4], 3985 kb/s, 25 fps, 25 tbr, 25 tbn
Input #1, mov,mp4,m4a,3gp,3g2,mj2, from 'dis.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    encoder         : Lavf59.3.100
  Duration: 00:00:01.64, start: 0.000000, bitrate: 1574 kb/s
  Stream #1:0(und): Video: h264 (High 4:4:4 Predictive) (avc1 / 0x31637661), yuv444p, 720x576, 1568 kb/s, 25 fps, 25 tbr, 12800 tbn (default)
    Metadata:
      handler_name    : VideoHandler
      vendor_id       : [0][0][0][0]
Stream mapping:
  Stream #0:0 (mpeg4) -> scale2ref:ref
  Stream #1:0 (h264) -> scale2ref:default
  libvmaf -> Stream #0:0 (wrapped_avframe)
Output #0, null, to '/dev/null':
  Metadata:
    software        : Lavf59.3.100
    encoder         : Lavf59.3.100
  Stream #0:0: Video: wrapped_avframe, yuv420p(tv, progressive), 720x576, q=2-31, 200 kb/s, 25 fps, 25 tbn (default)
    Metadata:
      encoder         : Lavc59.1.101 wrapped_avframe
[Parsed_libvmaf_3 @ 0x6ae3740] libvmaf encountered an error, check log for details
Error while filtering: Invalid argument
Failed to inject frame into filter network: Invalid argument
Error while processing the decoded data for stream #0:0
Conversion failed!

**Hope you can help me solve this question.**

**Comment by slhck** [comment]
I have to admit, it's hard with the libvmaf errors to know what the actual problem is. My guess is that it's using libvmaf 1.x and the model is a JSON model, which requires libvmaf 2.x.

Please check the section on the VMAF model: https://github.com/slhck/ffmpeg-quality-metrics#specifying-vmaf-model

If you downloaded the static build from johnvansickle.com, make sure to install the `.pkl` model files, too. You can then try:

```
ffmpeg_quality_metrics dist.mkv ref.mkv -m vmaf --model-path /usr/local/share/model/vmaf_v0.6.1.pkl
```

Assuming you've put them in `/usr/local/share/model`.

**Comment by littlewhitesea** [comment]
Thanks for your quick reply, I have sloved this problem. 

Followed your suggestion, I designated the "model-path", but the file 'vmaf_v0.6.1.pkl' in your porject didn't work in my computer. Finally, I found the file 'vmaf_float_v0.6.1.pkl' in https://github.com/Netflix/vmaf/tree/master/model is useful.

I put the project [vmaf-master](https://codeload.github.com/Netflix/vmaf/zip/refs/heads/master) in ```/media/```. Then, I obtained the right output when I performed the following command.
```
ffmpeg_quality_metrics dis.mp4 ref.mp4 -m vmaf --model-path /media/vmaf-master/model/vmaf_float_v0.6.1.pkl
```
The version of Python in my environment is 3.8.5.
